src/mcse.jl

```diff
@@ -23,6 +25,9 @@ function mcse(x::AbstractVector{<:Real}; method::Symbol=:imse, kwargs...)
         throw(ArgumentError("unsupported MCSE method $method"))
     end
 end
+function mcse(x::AbstractArray{<:Real,3}; kwargs...)
+    return dropdims(mapslices(xi -> mcse(vec(xi); kwargs...), x; dims=(2, 3)); dims=(2, 3))
```

Comment:
IIRC `mapslices` is quite inefficient and usually not very performant. I wonder if there's a performance problem here and, if yes, if it could be fixed?
Maybe also just use
```suggestion
    return vec(mapslices(xi -> mcse(vec(xi); kwargs...), x; dims=(2, 3)))
```
?

Reply:
Probably. `mapslices` is also not type-inferrable. We might be more efficient by using `eachslices` and constructing the output container ourselves.

I prefer `dropdims` here instead of `vec` because it's more friendly with named dimensions. e.g. this currently works:

```julia
julia> using MCMCDiagnosticTools, DimensionalData

julia> da = DimArray(randn(3, 10, 4), (Dim{:var}(1:3), Dim{:draw}(1:10), Dim{:chain}(1:4)));

julia> mcse(da)
3-element DimArray{Float64,1} with dimensions: 
  Dim{:var} Sampled{Int64} 1:3 ForwardOrdered Regular Points
and reference dimensions: 
  Dim{:draw} Sampled{Int64} 1:1 ForwardOrdered Regular Points,
  Dim{:chain} Sampled{Int64} 1:1 ForwardOrdered Regular Points
 1  0.187351
 2  0.146984
 3  0.159
```
In general our functions won't preserve named dimensions, but when they trivially can, it makes sense to do so.

Reply:
I see, yes, that's a valid argument against `vec`.

Reply:
I replaced this with an explicit allocation of the output container using the same type heuristic as `ess_rhat`, followed by a use of `eachslice` to fill.